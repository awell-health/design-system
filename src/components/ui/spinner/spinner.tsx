import { cva, type VariantProps } from 'class-variance-authority';

import { cn } from '@/lib/utils';

const spinnerVariants = cva('loading loading-spinner text-blue-600', {
  variants: {
    variant: {
      circle: 'loading-spinner',
      dots: 'loading-dots',
      infinity: 'loading-infinity'
    },
    size: {
      xs: 'loading-xs',
      sm: 'loading-sm',
      md: 'loading-md',
      lg: 'loading-lg'
    }
  },
  defaultVariants: {
    variant: 'circle',
    size: 'lg'
  }
});

export interface Props
  extends React.HTMLAttributes<HTMLDivElement>,
    VariantProps<typeof spinnerVariants> {}

function Spinner({ size, variant, className, ...props }: Props) {
  return <span className={cn(spinnerVariants({ size, variant }), className)} {...props} />;
}

export { Spinner };
