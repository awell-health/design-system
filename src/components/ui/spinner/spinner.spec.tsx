import { describe, expect, it } from 'vitest';

import React from 'react';
import { Spinner } from './spinner';
import { render } from '@testing-library/react';

describe('Spinner', () => {
  it('match snapshot', () => {
    const { container } = render(<Spinner />);

    expect(container).toMatchSnapshot();
  });
});
