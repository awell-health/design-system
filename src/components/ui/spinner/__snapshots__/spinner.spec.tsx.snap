// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Spinner > match circle snapshot 1`] = `
<div>
  <span
    class="loading loading-spinner text-blue-700 loading-spinner loading-lg"
  />
</div>
`;

exports[`Spinner > match dots snapshot 1`] = `
<div>
  <span
    class="loading loading-spinner text-blue-700 loading-dots loading-lg"
  />
</div>
`;
